src/Magnum/MeshTools/BoundingVolume.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019,
+                2020, 2021, 2022 Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2022 Pablo Escobar <mail@rvrs.in>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "BoundingVolume.h"
+
+#include <Corrade/Containers/StridedArrayView.h>
+
+#include "Magnum/Math/FunctionsBatch.h"
+
+namespace Magnum { namespace MeshTools {
+
+Range3D boundingBoxAxisAligned(const Containers::StridedArrayView1D<const Vector3>& points) {
+    const std::pair<Vector3, Vector3> minmax = Math::minmax(points);
+    return {minmax.first, minmax.second};
+}
+
+Corrade::Containers::Pair<Vector3, Float> boundingSphereBouncingBubble(const Containers::StridedArrayView1D<const Vector3>& points) {
+    /* See comment about radius below, this is done for consistency */
+    if(points.isEmpty()) return {{}, Math::TypeTraits<Float>::epsilon()};
+
+    /** @todo Skip NaNs here? To match behaviour of boundingBoxAxisAligned()
+        which uses minmax(). */
+    Vector3 center = points[0];
+    /* Radius ends up in the denominator in the first loop so we can't
+       initialize it to 0. Unfortunately this also means the returned radius is
+       always above or equal to epsilon. */
+    Float radius = Math::TypeTraits<Float>::epsilon();
+    Float radiusSquared = radius*radius;
+
+    for(int i = 0; i < 2; ++i) {
+        for(const Vector3& p : points) {
+            const Float ds = (p - center).dot();
+            if(ds > radiusSquared) {
+                const Float alphaInv = radius/Math::sqrt(ds);
+                const Float alphaSqInv = alphaInv*alphaInv;
```

Comment:
Wouldn't this lead to less precision loss, maybe? Not sure how important tho (and I got no experience with testing numerical stability of algorithms either).

```suggestion
                const Float alphaSqInv = radius*radius/ds;
```

Reply:
I... don't really know. My intention here was to avoid a division. Let me read up on that a little bit.

Reply:
Not sure if this idea is helpful at all, but I suppose the smaller precision loss could make the calculated output more stable when tried on randomly shuffled position arrays. Don't know how to measure that better :sweat_smile:

Reply:
Done 😊 I was overthinking this a little bit